OTAPI.Scripts/Mods/HookNpcBossBag.Server.cs

```diff
@@ -132,14 +109,20 @@ Terraria.NPC npc
                 };
                 BossBag?.Invoke(null, args);
                 if (args.Result == HookResult.Cancel)
-                    return -1;
+                {
+#if TerrariaServer_EntitySourcesActive || Terraria_EntitySourcesActive || tModLoader_EntitySourcesActive
+                    return Terraria.Item.NewItem(Source, args.X, args.Y, args.Width, args.Height, args.Type, args.Stack, args.NoBroadcast, args.Pfix, args.NoGrabDelay, args.ReverseLookup);
+#else
+                    return Terraria.Item.NewItem(args.X, args.Y, args.Width, args.Height, args.Type, args.Stack, args.NoBroadcast, args.Pfix, args.NoGrabDelay, args.ReverseLookup);
+#endif
+                }
```

Comment:
i may not be up to scratch with these hooks nowadays, but if it was cancelled, should a new item be created still? And if so, why?

Reply:
According to how "HookResult.Cancel" tend to do, we should not let the new item be created, and i completely agree with that. Sorry, it was because of my lack of knowledge that i overlooked it.